wordpress-mobile/WordPress-Android
# QR Code Auth: Implement flow

Parent #16481 

This PR is the cumulation of 8 separate PRs. See the list in the parent issue under the heading **Implement Auth Flow**
Each PR was tested individually, but I recommend a full end-to-end test.

When adding the dependencies for the Google Scanner it was noted that some transient dependencies have been updated (see this wpmobilebot [comment](https://github.com/wordpress-mobile/WordPress-Android/pull/16710#issuecomment-1146482877)). As a result, there are some sanity checks to be run during the testing phase to verify I didn't screw anything else up. I have also run this test before submitting this PR.

**Prior to testing this PR you will need to**
- Make sure your real device is sandboxed
- Make sure your sandbox is pointed to D79766-code (apply the patch)
- Have access to the "Calypso live (direct link)" in the related Calypso PR: https://github.com/Automattic/wp-calypso/pull/63257

**To test:**
Pre-req
- Install the app
- Login using a WP.com account (not an automattic account or one with 2FA)
- Navigate to Me -> App Settings -> Debug Settings
- Enable QRCodeAuthFlowFeatureConfig
- Restart the App

**Testing from the Scan Login code menu item**
- Launch the app
- Navigate to My Site > Me > Scan Login Code
- Scan the QR Code from the "Calypso live (direct link)" in the Calypso PR (See above). **Make sure you are not logged in to WP.com when you go to the direct link. If you use an incognito window to open the link, the you can't verify the last item.
- ✅ Verify you see a "Are you trying to login" message that displays your web browser and approximate location
"Are you trying to login to Chrome near Los Angeles, California?"
- Tap the 'Yes, log me in' button
- ✅ Verify you see a loading indicator with the current view faded out
- ✅ Verify after a second you see the "You're logged in!" message
- ✅ Verify tapping Dismiss exits the flow
- ✅ Verify you are logged in on your web browser

**Sanity Checks**
- Remove any other versions of the WP app from your device
- Install the app and login 
- Test that you can still receive notifications (update a post on a different device)
- Test that you can still set/receive blogging reminder notifications

## Regression Notes
1. Potential unintended areas of impact


2. What I did to test those areas of impact (or what existing automated tests I relied on)


3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
